src/ansys_sphinx_theme/cheatsheet.py

```diff
@@ -0,0 +1,228 @@
+
+logger = logging.getLogger(__name__)
+
+# Cheat sheet extension version
+CHEAT_SHEET_QUARTO_EXTENTION_VERSION = "v1"
+
+
+def cheatsheet_sidebar_pages(app: Sphinx) -> Optional[List[str]]:
+    """
+    Get the pages to display the cheat sheet sidebar.
+
+    Parameters
+    ----------
+    app : sphinx.application.Sphinx
+        Application instance for rendering the documentation.
+
+    Returns
+    -------
+    Optional[List[str]]
+        List of pages to display the cheat sheet sidebar, or None if not configured.
+    """
+    html_theme_options = app.config.html_theme_options
+    cheatsheet_options = html_theme_options.get("cheatsheet")
+    if not cheatsheet_options:
+        return None
+    pages = cheatsheet_options.get("pages", ["index"])
+    cheatsheet_pages = [pages] if isinstance(pages, str) else pages
+    return cheatsheet_pages
+
+
+def convert_pdf_to_png(pdf_path: pathlib.Path, output_dir: pathlib.Path, output_png: str) -> None:
+    """
+    Convert PDF to PNG images.
+
+    Parameters
+    ----------
+    pdf_path : pathlib.Path
+        Path to the PDF file.
+    output_dir : pathlib.Path
+        Path to the output directory.
+    output_png : str
+        Name of the output PNG file.
+    """
+    try:
+        from pdf2image import convert_from_path
+    except ImportError as e:
+        raise ImportError(
+            f"Failed to import `pdf2image`: {e}. Install the package using `pip install pdf2image`"
+        )
+    try:
+        images = convert_from_path(pdf_path, 500)
+        images[0].save(output_dir / output_png, "PNG")
+    except Exception as e:
+        raise RuntimeError(
+            f"Failed to convert PDF to PNG: {e}. Ensure the PDF file is valid and poppler is installed."  # noqa:E501
+        )
+
+
+def run_quarto_command(command: List[str], cwd: str) -> None:
+    """
+    Run a Quarto command and log its output.
+
+    Parameters
+    ----------
+    command : List[str]
+        List of command arguments.
+    cwd : str
+        Current working directory.
+    """
+    command = ["quarto"] + command
+    try:
+        result = subprocess.run(command, cwd=cwd, check=True, capture_output=True, text=True)
+        if result.stdout:
+            logger.info(result.stdout)
+
+        if result.stderr:
+            logger.error(result.stderr)
+
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError(f"Failed to run the command: {e}")
+
+
+def add_cheat_sheet(
+    app: Sphinx,
+    pagename: str,
+    templatename: str,
+    context: Dict[str, Any],
+    doctree: Optional[document],
+) -> None:
+    """
+    Add a cheat sheet to the left navigation sidebar.
+
+    Parameters
+    ----------
+    app : sphinx.application.Sphinx
+        Application instance for rendering the documentation.
+    pagename : str
+        Name of the current page.
+    templatename : str
+        Name of the template being used.
+    context : Dict[str, Any]
+        Context dictionary for the page.
```

Comment:
Let's elaborate more about this description.

Reply:
sure adding it